### ISSUE TITLE
live server not reloading automatically

### ISSUE THREAD

**Issue by hillarion99** [ritwickdey/vscode-live-server#452]
## I'm submitting a...

```html
[x] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Feature request
[x] Documentation issue or request

## Current behavior

live server not reloading automatically

## Expected behavior

it reload without manually reloading the browser

## Environment

```html
Browser:

- [x] Chrome (desktop) version 
- [x] Firefox version 

For Tooling issues:

- Live Server: 5.6.1
- Platform:  Windows
- Visual Studio Code: 1.33.1

**Comment by hillarion99** [comment]
Can anybody please answer me???

**Comment by pordemoni** [comment]
Same goes for me. I noticed it recently. Live server would serve the file in the browser, but it would **not refresh automatically upon saving the file**.

**Comment by pordemoni** [comment]
**SOLVED**

In my case:
I had my HTML file inside a folder, which name starts with a period `.`

Don't do this 👇 
> http://127.0.0.1:5500/.folder/index.html

**Comment by alfarians** [comment]
I'm experiencing the same problem here.
I think the recent VSCode update made it unworking as expected.
Before updating, it worked well.

**Comment by jeffus07** [comment]
> I'm experiencing the same problem here.
> I think the recent VSCode update made it unworking as expected.
> Before updating, it worked well.

Yes, the recent update has broken it

**Comment by eletitos** [comment]
Hi, I have the same problems.
Auto-refreshing isn´t working.

My enviroment:
Windows 10.
VSCode 1.35.1 (I have also tried in 1.34 and it didn't work either)
Live server: 5.6.1
Browser: Chrome aor Firefox (I have tried in both of them)

I have tried re-installing, and deleting the files from extension file for a completed resetting. 
Any solution?

**Comment by dijiflex** [comment]
## I too have the same problem . Live reload not working
Chrome Version: 75.0.3770.100 (Official Build) (64-bit)
Vscode version 1.36
Live server version 5.6.1

**Comment by stevekb** [comment]
I had the same issue and my problem was that my html file wasn't encoded in utf8

**Comment by danivideda** [comment]
I had the same issue. Until now, i couldn't figure out what's the problem. My environment:
Windows 10
Chrome Version: 76.0.3809.100 (Official Build) (64-bit)
VSCode 1.36.1
Live Server 5.6.1

So far, it's been a random occurrence for me when i encounter this "page doesn't reload automatically". I sometimes have it working, but only a couple of minutes until the "auto-reload" doesn't work anymore. When i restarted my computer, it's a 50/50 chance that LiveServer will work correctly. Right now, i dont have a reliable solution as what may causing this problem. I think it could be my Windows 10, but i don't know how to check it.

**Comment by anemish1996** [comment]
Make sure to save your workspace configuration then it will work.

**Comment by Dranzule** [comment]
Having the same problem after updating to 1.37.1, tried deleting  ritwickdey.liveserver-5.6.1 from C:\Users\USERNAME\extensions\ritwickdey.liveserver-5.6.1, and re-installing VS CODE, but not working yet as well.

**Comment by swiftouch** [comment]
Same problem. It doesn't auto-refresh in the browser. It is nice it launches it in a browser window but I miss the auto-refresh. I can't get Settings Sync to work either. I thought maybe it's because I used the "system installer" instead of the user installer. Nope. Neither work. So far this week, I've spent more time trying to get extensions working than I have coding. :*(

**Comment by donnyv** [comment]
It only refreshes after I save the doc for me.

**Comment by positesitesite** [comment]
> It only refreshes after I save the doc for me.
Had the same issue. Was solved by changing autosave option in setting to afterDelay.

**Comment by StarBright3** [comment]
Make sure <meta charset = "UTF-8"> is in the head.

**Comment by moghaazi** [comment]
> Make sure meta charset = "UTF-8" is in the head.

Worked For Me, Thanks Man

**Comment by tyler-ray** [comment]
Works on my work computer, not my personal computer. <meta charset="UTF-8"/>, refreshed, reloaded, uninstalled & re-installed several different ways. Still not auto-refreshing.  Console says "live reload enabled"... I've narrowed my open folders to just one,...

**Comment by tjbulick** [comment]
> I had the same issue and my problem was that my html file wasn't encoded in utf8

thanks bro, it worked

**Comment by jefflundberg** [comment]
I was having the same issue. 

It seems to work fine if I open _a specific file_ with Live Server by either the `Ctrl-Shift-P` Command Pallet, Right-click menu, or `Alt-L Alt-O` key command. It doesn't work properly if I start Live Server by clicking "Go Live" in the status bar at the bottom.

Hope this helps someone!

**Comment by kottuzik** [comment]
> I was having the same issue.
> 
> It seems to work fine if I open _a specific file_ with Live Server by either the `Ctrl-Shift-P` Command Pallet, Right-click menu, or `Alt-L Alt-O` key command. It doesn't work properly if I start Live Server by clicking "Go Live" in the status bar at the bottom.
> 
> Hope this helps someone!

This really works.

**Comment by vronney** [comment]
Anyone able to get Live Server to update in Chrome?

**Comment by kottuzik** [comment]
Mine is working.

**Comment by iwswordpress** [comment]
I really like this extension and have used it for a long time. Tonight it does not reload and I need to a hard refresh to get current page.

Reinstalled and checked for updates to no avail.

Would love to get it back working as I really like it!

Thanks for all the work you have put in.

Craig

**Comment by ArrestedMango** [comment]
Was having the same issue.  Looks like it was as simple as how I was opening the file.  When I go to File > Open Folder, it seems to work every time.  When I open the individual files, it doesn't work.  This thread was the key, Thank you!

**Comment by hcientist** [comment]
I just found this, and then randomly at the same time I noticed a notification in vscode telling me that live reload isn't possible without `head` and `body` (I was editing a file that's just an html fragment)

**Comment by sonicemma** [comment]
I've been having the issues on two different computers, both using Windows 10. 

Both times, my CSS file was not updating. My html file was updated, but oddly enough, not CSS.

**Comment by yednaphsotnas** [comment]
There's one thing that resolves the issue.
• Open settings.
• Search for 'autosave'.
• Set it to 'afterDelay'.
• Just below that, set the AutoSaveDelay to anything very less like 5 (I have set it to 1)
• Now search for 'Wait' in settings.
• In LiveServer > Settings:Wait, set the value very low (e.g. 3).
This should help regardless of platforms :)

**Comment by ceign** [comment]
I have the same problem, like @pordemoni , the name of my folder starts with '#', change the start character for a letter and the problem was solved.

**Comment by felipisan** [comment]
@yednaphsotnas Thanks, dude.

**Comment by kaizen10times** [comment]
I NGINX server and able to load my website using proxy, but it does reload if i make change & save in a html.twig file. Please help!

**Comment by wilsonrg** [comment]
Live Server não atualiza após salvar arquivo html via vscode

Versões:
- Live Server v5.6.1
- VSCode v1.42.1

Tentativas de reparo:

- Desativado "onSave" para "off"
- Reinstalado Live Server
- Restart no VSCode

- salvamento automático: afterDelay
- AutoSaveDelay: 1
- LiveServer Wait: 1


Alguém tem uma solução por favor?

**Comment by lchugh74** [comment]
> There's one thing that resolves the issue.
> • Open settings.
> • Search for 'autosave'.
> • Set it to 'afterDelay'.
> • Just below that, set the AutoSaveDelay to anything very less like 5 (I have set it to 1)
> • Now search for 'Wait' in settings.
> • In LiveServer > Settings:Wait, set the value very low (e.g. 3).
> This should help regardless of platforms :)

This worked for me..
thanks 💃

**Comment by icematthews** [comment]
> 
> 
> Was having the same issue. Looks like it was as simple as how I was opening the file. When I go to File > Open Folder, it seems to work every time. When I open the individual files, it doesn't work. This thread was the key, Thank you!

Thanks this works

**Comment by wilsonrg** [comment]
Meu Live Server voltou a funcionar após instalar outro plugin chamado Live Server Preview
Instalei, reiniciei o VSCode e testei o Live Server que funcionou corretamente.
Espero que outras pessoas possam ser ajudadas com esta solução estranha assim como eu fui.

**Comment by Strangenina** [comment]
> Meu Live Server voltou a funcionar após instalar outro plugin chamado Live Server Preview
> Instalei, reiniciei o VSCode e testei o Live Server que funcionou corretamente.
> Espero que outras pessoas possam ser ajudadas com esta solução estranha assim como eu fui.

O meu não funcionou de jeito nenhum. não atualiza sozinho como deveria ser.

**Comment by Go3shom** [comment]
> 
> 
> **SOLVED**
> 
> In my case:
> I had my HTML file inside a folder, which name starts with a period `.`
> 
> Don't do this 👇
> 
> > http://127.0.0.1:5500/.folder/index.html

Same here but mine was #folderName

**Comment by GabrielNicolasAvellaneda** [comment]
Looks like that live server fails to refresh if you have some issue with your html. In my case I had a silly mistake, I had a self-closing script tag `<script src="index.js" />`. Which Chrome recognized and properly loaded the javascript file but Live Server didn't

**Comment by wmilambiling** [comment]
guilty as charged! this worked for me.

**Comment by mdrasheduzzamanakash** [comment]
ensure that you have written the html in proper way.
<html>
<body>
    ....
</body>
</html>

**Comment by mdrasheduzzamanakash** [comment]
make sure that you have written 
html code with <html> tag.

**Comment by mdrasheduzzamanakash** [comment]
make sure you have written html code with html tag not just body elements. Sometimes it doesn't work without proper html format.

**Comment by elspnm** [comment]
with windows 7 and recent chrome v81 works fine but in windows 10 it is not launching the browser.
Tried to run with compatibility mode still not launching.

Only shows port info but browser not opening at all

**Comment by Chi-coder** [comment]
This is so frustrating especially for a beginner trying to get a grasp on things. The extension suddenly stopped working. I've deleted, reinstalled, followed the settings doc to configure the extension in settings.json but nothing. Clearly there's something interfering with the software running in the browsers, because right in the console it shows the Live Server icon and even runs but doesn't open the browsers. 🤦🏽‍♂️
![15899222299953776176068150227381](https://user-images.githubusercontent.com/63807877/82378281-c4383080-9a1c-11ea-9659-bf533502f080.jpg)
![15899223855033952867499278097342](https://user-images.githubusercontent.com/63807877/82378583-3dd01e80-9a1d-11ea-95fb-725a715f7c39.jpg)

**Comment by Darkmift** [comment]
Pretty sure its safe to say this repo is no longer maintained by the creator.

**Comment by MarkTRoss** [comment]
In my case, I did have it under a path called _Notes and moving it out made not difference.
I did'ht have the html/head/utf-8 set as I was only play with code and below fix it.

**<!doctype html>
<html>
<head>
<meta charset="utf-8">
..
..
</head>
</html>**

**Comment by Falencar-dev** [comment]
> 
> 
> There's one thing that resolves the issue.
> • Open settings.
> • Search for 'autosave'.
> • Set it to 'afterDelay'.
> • Just below that, set the AutoSaveDelay to anything very less like 5 (I have set it to 1)
> • Now search for 'Wait' in settings.
> • In LiveServer > Settings:Wait, set the value very low (e.g. 3).
> This should help regardless of platforms :)

This solved my problem! Thank you so much.. I've been trying to find a solution for a while now.. After i changed the following settings it works like magic for me auto reloading..
Thanks!

**Comment by timjanssens** [comment]
When i start the live server with windows i am able to see the changes. When i switch to remote-WSL the site doesn't update anymore when i save.  Going back to "reopen folder in Windows" and the site is updating again.
Any solution for this?

**Comment by phos23** [comment]
I never could get autorefresh to work, tried opening through 'Go Live button, RMB, changing VSC settings to autosave auto delay, changing liveserver delay to 5, (html already in utf-8) nothing has worked, I always had to refresh it manually in the browser.
 
Problem was **Firefox**, when I tried in chrome, everything works as it supposed to be, not sure where the problem is though...

**Comment by noobato1** [comment]
**SOLVED** (For me at least)

In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

**Comment by MaceGood** [comment]
@noobato1 thank you bro, thanks for help, I LOVE YOU <3!

**Comment by deramirez1** [comment]
I was able to fix this by changing to a different distros. I was running WSL2-Ubuntu 20.04 with another workspace. I had switched workspaces to start another project but I was stilling running on the WSL distro. Steps I took to change it:

- Open remote window

**Comment by santimirandarp** [comment]
In my case, it there was some trouble with firefox dev edition. Just use it on classic firefox and works fine.

**Comment by AleDenigres** [comment]
**SOLVED** (for me at least)

I installed the _Live Server_ **Chrome extension** in my browser, enabled the _Live Reload_ setting and reloaded the page once. From that moment on, the automatic reload started working! Hope it helps!

**Comment by nemostudent** [comment]
> **SOLVED** (for me at least)
> 
> I installed the [_Live Server_ **Chrome extension**](https://chrome.google.com/webstore/detail/live-server-web-extension/fiegdmejfepffgpnejdinekhfieaogmj) in my browser, enabled the _Live Reload_ setting and reloaded the page once. From that moment on, the automatic reload started working! Hope it helps!

Didn't help :(

**Comment by Pannu786** [comment]
wow, so no one actually has a any solid solution for this yet ??

**Comment by Darkmift** [comment]
Get npm live-server.
Works great...if you have node 😃

**Comment by MarkTRoss** [comment]
What does the Live Server Chrome extension do?

One thing I remember is that if you rename the extension folder under Programdata - no good, you have to delete it or move it out.

I would not be able to live without this extension, this and his SASS Live compiler [Ritwick Dey - I owe you a beer]

**Comment by Pouria-R** [comment]
Hi guys. it's simple. if you go to setting you will see auto save is off. choose "after delay" and done.

**Comment by stefanrows** [comment]
Ok guys, I figured out what my issue was and maybe it can help someone else.

I tried everything in this thread, I switches browsers and still Auto Reload wouldn't work. Then it dawned on me. I use WSL2. 

Immediately I thought that might be the culprit. Deleted my repo from my WSL directory and cloned it again directly into my Windows 10 Documents folder. And it immediately worked.

Tested on both FF and Chrome.

Hope that helps.

**Comment by Pannu786** [comment]
Well my issue was I gave my folder name start with "#".. Don't ask why I
did that but apparently if you use any special characters for a folder name
then it also doesn't work.

**Comment by kamzard** [comment]
I fix this just activating Use local ip and Full Reload in configuration

**Comment by theking2** [comment]
Does not work on VSCode 1.49.2 Live Server 5.6.1
```
[Extension Host] Change detected d:\Cloud\index.html
console.ts:137 [Extension Host] Looking for parseable documents...
console.ts:137 [Extension Host] Found all parseable documents.
console.ts:137 [Extension Host] Parsing documents and looking for CSS class definitions...
console.ts:137 [Extension Host] Summary:
console.ts:137 [Extension Host]% 1 parseable documents found
console.ts:137 [Extension Host]% 0 CSS class definitions found
console.ts:137 [Extension Host]% 0 unique CSS class definitions found
console.ts:137 [Extension Host]% 0 failed attempts to parse. List of the documents:
console.ts:137 [Extension Host] 
```

**Comment by theking2** [comment]
> 
> 
> Does not work on VSCode 1.49.2 Live Server 5.6.1
> 
> ```
> [Extension Host] Change detected d:\Cloud\index.html
> console.ts:137 [Extension Host] Looking for parseable documents...
> console.ts:137 [Extension Host] Found all parseable documents.
> console.ts:137 [Extension Host] Parsing documents and looking for CSS class definitions...
> console.ts:137 [Extension Host] Summary:
> console.ts:137 [Extension Host]% 1 parseable documents found
> console.ts:137 [Extension Host]% 0 CSS class definitions found
> console.ts:137 [Extension Host]% 0 unique CSS class definitions found
> console.ts:137 [Extension Host]% 0 failed attempts to parse. List of the documents:
> console.ts:137 [Extension Host] 
> ```

The reason was a missing <head></head> in the html file. If no <head> section is available in the source for the live-server cannot be injected. According to https://www.w3.org/MarkUp/html3/dochead.html the head section can be omitted. So this appears to be a bug in Live Server 5.6.1.

**Comment by davidmatthew-dev** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

This worked for me 9/29/2010 - thanks!

**Comment by drnourhomsi** [comment]
For me the problem was that I am running another website on the same port.
Just make sure you close VSC or and other program that is using ports like: Local or MAMP

**Comment by enqs4h** [comment]
> > It only refreshes after I save the doc for me.
> 
> Had the same issue. Was solved by changing autosave option in setting to afterDelay.

This worked for me. thank you!!!

**Comment by yulose9** [comment]
The problem still persist tho

**Comment by yulose9** [comment]
> The problem still persist tho

Found a fix, I've recently enabled _Format on Save_ and turning it of resolves my issue

**Comment by jaysuthar743** [comment]
I had the same issue on Linux ubuntu and I resolved it by changing the number of files vsCode tracker can track:

open file: `/etc/sysctl.conf`

add following line at the end: 
**`fs.inotify.max_user_watches=524288`**

close all folder,
reopen vsCode,
open folder,
Click on file,
right-click-> open with live server.

**Comment by ratsuns** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

That did the TRICK!!!

**Comment by albas99** [comment]
> **SOLVED**
> 
> In my case:
> I had my HTML file inside a folder, which name starts with a period `.`
> 
> Don't do this 👇
> 
> > http://127.0.0.1:5500/.folder/index.html

I've been on this issue for five days straight, I never thought it was something as trivial as this

**Comment by ehsanajmal** [comment]
can anyone help me please?
i tried all the solution but no one is working for me :(
the problem is that the live server extension is not working on saving the file, it only works when i go and refresh the browser page
can anyone help me with this please
thanks in advance

**Comment by MarkTRoss** [comment]
Is your html code correctly formatted, if I just do code samples without the correct head tag - it wont work.
..and maybe you also need the <!DOCTYPE html>.

**Comment by itsraghz** [comment]
Hi friends, it worked fine for me in the beginning, until I installed a few more extensions (like Browser Preview) and also played around with the settings. Later I made more confusions on the settings on delayAfterSave, wait etc., as the live server stopped working. A restart of the vscode has not helped me at all.

After googling a bit and also on this thread, I started redoing the things. I opened the "settings.json"  file in the VSCode and removed all the settings related to browser preview, and it worked again :) May be you can try resetting the entries and toggle the entries to find out what is dominating what. Please be informed that a few extensions might dominate the similar/relevant entries.

FYI, my settings.json as of now (seems working fine) copy is below.

`{
	"workbench.colorTheme": "Solarized Dark",
	"editor.minimap.enabled": false,
	"editor.formatOnSave": true,
	"editor.defaultFormatter": "esbenp.prettier-vscode",
	"prettier.useTabs": true,
	//"liveServer.settings.useWebExt": true,
	//"liveServer.settings.fullReload": false,
	"liveServer.settings.donotShowInfoMsg": false,
	"liveServer.settings.CustomBrowser": "chrome" //,
	//"liveServer.settings.wait": 1,
	//"files.autoSave": "afterDelay",
	//"files.autoSaveDelay": 1
}
`

The lines commented were played one after another to bring back the live server. I think the default value of 100 ms is still fine for the wait if any in looking for. Hope it helps someone else.

Cheers,
RM.

**Comment by EMcGitBsa** [comment]
> Hi guys. it's simple. if you go to setting you will see auto save is off. choose "after delay" and done.

@Pouria-R

May have been a combination of setting up the Live Server Chrome extension in my browser and enabling the Live Reload setting but neither worked until checking this option.  Thanks!!!

**Comment by mattrmay** [comment]
Hi All!

It appears that the issue I'm having is slightly different and I'm not sure if I need to open a new issue or not.

**What's Expected:** The browser to auto-reload/refresh on file edit.
**What's Happening:** The browser will only auto-reload when the edited file is saved.
**Ask:** Is this normal behavior or is it correct to assume that I should see edits appear in "real time"? If the expected behavior is correct, what can I try to correct this issue?

Visual Studio Code
```
Version: 1.52.0 (user setup)
Commit: 940b5f4bb5fa47866a54529ed759d95d09ee80be
Date: 2020-12-10T22:45:11.850Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19042
```

Live Server
```
Id: ritwickdey.liveserver
Version: 5.6.1

"liveServer.settings.AdvanceCustomBrowserCmdLine": null
```

Thank you in advance to anyone who is able to answer this question.

-Matt

**Comment by MarkTRoss** [comment]
for me anyways and how I believed it to work is.
after I save, that's when the browser auto-refreshes.
(not when I type changes).

**Comment by Esh07** [comment]
Somehow this configuration worked for me. 
Live server able to open the webpage on the browser but it was not reloading automatically after i saved the file, so i tried all the above methods that stated by the user but unfortunatlly none of them work for me. 
Then i follow this step: [#20](https://github.com/ritwickdey/live-server-web-extension/issues/20#issuecomment-615876586)
and it worked. 
just go to **Setting** , serach for **Use Local Ip** and tick the checkbox.

**Comment by aryanaruni** [comment]
Can anyone help me please?
I noticed it recently. Live server would serve the file in the browser, but it would not refresh the browser automatically upon saving the file.
Live Server: v5.6.1
VS Code:  Version: 1.52.1 (user setup).
I re-installed VS Code and Live Sever but the problem is not solved.

**Comment by Esh07** [comment]
> Can anyone help me please?
> I noticed it recently. Live server would serve the file in the browser, but it would not refresh the browser automatically upon saving the file.
> Live Server: v5.6.1
> VS Code: Version: 1.52.1 (user setup).
> I re-installed VS Code and Live Sever but the problem is not solved.

This worked for me.
[Here] (https://github.com/ritwickdey/vscode-live-server/issues/452#issuecomment-747425741)

**Comment by Banoulka** [comment]
> > **SOLVED** (For me at least)
> > In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!
> 
> Clicked the checkbox and that did the TRICK!!!

Spent ages on this thread trying with different settings, unchecking this worked for me

**Comment by ASTRO-N1** [comment]
> Make sure meta charset = "UTF-8" is in the head.

Its there in my head tag but its still **not** **auto-refreshing** why is this so??

**Comment by ASTRO-N1** [comment]
> ## I'm submitting a...
> ```
> [x] Regression (a behavior that used to work and stopped working in a new release)
> [x] Bug report  
> [ ] Feature request
> [x] Documentation issue or request
> 
> ## Current behavior
> 
> live server not reloading automatically
> 
> ## Expected behavior
> 
> it reload without manually reloading the browser
> 
> ## Environment
> 
> ```html
> Browser:
> 
> - [x] Chrome (desktop) version 
> - [x] Firefox version 
> 
> For Tooling issues:
> 
> - Live Server: 5.6.1
> - Platform:  Windows
> - Visual Studio Code: 1.33.1
> ```

Ok so I tried doing one thing and it worked !

**So Just Check If Your Folder Name Starts With Any Thing Like This : . # @ ! or anything else just rename that and remove any marks from the folder name**

Mine was #JsSite and thats why it was not auto reloding but now it works 🥇

**Comment by AmrulHarahap** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

Finally.. This is work for me. Thanks a lot

**Comment by ksparb** [comment]
If nothing else here works for you; if you have a browser extension that by default blocks JavaScript until you whitelist the page (in my case ScriptSafe), make sure that you've actually whitelisted 127.0.0.1 or whichever IP you use for Live Server.

**Comment by cesaralmeidareyes** [comment]
> > **SOLVED** (For me at least)
> > In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!
> 
> Finally.. This is work for me. Thanks a lot

This worked for me.

Thanks!

**Comment by Arramis** [comment]
For me it refreshes only if i save. (linux mint 20.1 ulysa) - 

Version: 1.53.0-insider
Commit: 861a88ebadb4335df92251faf6420b598440e082
Date: 2021-01-29T21:49:47.914Z
Electron: 11.2.1
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Linux x64 5.4.0-65-generic)

**Comment by KevinT27** [comment]
> Make sure meta charset = "UTF-8" is in the head.

Lol this was the whole solutin

**Comment by ivancardozo11** [comment]
If you are here couse live server is not reloading on Brave browser but it is on Chrome , try to disable Brave shields (the brave icon that disable scripts and that stuff...)

It work for me :p

**Comment by YatinHasija** [comment]
I have seen this issue in a lot of machines, It generally happens when we have lot of things installed such as neon, anaconda etc.
Best you can do is create a new system user and use vs code over there so it won't inter-fare with other environment variables and work perfectly.

**Comment by Arramis** [comment]
I found out an workaround. 
Steps to reproduce:
1 . I installed LiveServer extension for chrome. 
2. Closed browser.
3. Start LiveServer in VSCode.
4. Open browser, start server extension.
Works perfectly.
Note: Sometime i need to push "live reload" button to switch it off  on chrome extension and restart chrome. But it works.
My computer probably has a lot of issues and I`m not in the mood to reinstall everything yet :).

Hope it helps.

**Comment by JoaquinGodoy97** [comment]
For me, I deleted "code" folder which was in Roaming. Then when I reinstalled VSC the firewall asked me the permission for a Port I believe, I can't remember well, but it was right after I open the Live Server tab. Also before that, I restarted some firewall settings. It has to do with the _firewall and the browser_  you're using in the case is none of the solutions you have given. I hope this could help you.

**Comment by DonThePhan** [comment]
Seems like it's just not working for me because I didn't have the html boiler plate. Once I had the bare bones, all seemed to be up and running. Guessing this isn't the same issue most are dealing with here?

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    !!!! This live updates!!!!
</body>
</html>
```

**Comment by MarkTRoss** [comment]
I've been caught with not having a html boiler plate.
The other day it stop working so I uninstalled and delete from the vscode extension folder C:\Users\{me}\.vscode\extensions, then reinstall it.

**Comment by Nihal-prog** [comment]
Guys This Worked for me :

Go to your browser then go to settings and check if JavaScript is Enabled

:)

**Comment by mpagels** [comment]
> If nothing else here works for you; if you have a browser extension that by default blocks JavaScript until you whitelist the page (in my case ScriptSafe), make sure that you've actually whitelisted 127.0.0.1 or whichever IP you use for Live Server.

This helped me. I tried all the other suggestions but nothings worked. But I have this JavaScript Blocker extension... After whitelisted the page, the reload is working now. Thx.

**Comment by czanon** [comment]
Don't breake your head....Put in VScode (in settings) the AutoSave option with a AfterDelay of 600 ms(for example) and you will have your automatic refresh on Live Server while you are editing...

**Comment by shoshonearnold** [comment]
> Don't breake your head....Put in VScode (in settings) the Auto Save option with a AfterDelay of 600 ms(for example) and you will have your automatic refresh on Live Server while you are editing...
> There are Youtube videos on that "issue"... is more simple of what you think...

This was my issue and once i did this on my new mac with vs code installed it was fixed. I did nealry ruin my whole afternoon becuase i forgot to change this, thanks czanon.

**Comment by HarshilSharaf** [comment]
> Hi friends, it worked fine for me in the beginning, until I installed a few more extensions (like Browser Preview) and also played around with the settings. Later I made more confusions on the settings on delayAfterSave, wait etc., as the live server stopped working. A restart of the vscode has not helped me at all.
> 
> After googling a bit and also on this thread, I started redoing the things. I opened the "settings.json" file in the VSCode and removed all the settings related to browser preview, and it worked again :) May be you can try resetting the entries and toggle the entries to find out what is dominating what. Please be informed that a few extensions might dominate the similar/relevant entries.
> 
> FYI, my settings.json as of now (seems working fine) copy is below.
> 
> `{
> 	"workbench.colorTheme": "Solarized Dark"
> 	"editor.minimap.enabled": false,
> 	"editor.formatOnSave": true,
> 	"editor.defaultFormatter": "esbenp.prettier-vscode",
> 	"prettier.useTabs": true,
> 	//"liveServer.settings.useWebExt": true,
> 	//"liveServer.settings.fullReload": false,
> 	"liveServer.settings.donotShowInfoMsg": false,
> 	"liveServer.settings.CustomBrowser": "chrome" //,
> 	//"liveServer.settings.wait": 1,
> 	//"files.autoSave": "afterDelay",
> 	//"files.autoSaveDelay": 1
> }
> `
> The lines commented were played one after another to bring back the live server. I think the default value of 100 ms is still fine for the wait if any in looking for. Hope it helps someone else.
> 
> Cheers,
> RM.

IT worked for me Thanks 😉😉

**Comment by SHUBHAM-DHANUKA** [comment]
> Make sure meta charset = "UTF-8" is in the head.

Thanks, it is working great. no issue with autosaves and any other configuration.

**Comment by Matulaz** [comment]
> **SOLVED**
> 
> In my case:
> I had my HTML file inside a folder, which name starts with a period `.`
> 
> Don't do this 👇
> 
> > http://127.0.0.1:5500/.folder/index.html
---------------------------------------------------
Same thing but my folders name starts with a # ... Problem Solved

**Comment by ZacharyVe** [comment]
My problem went away as soon as I enable autosave in VS. It seems to be triggered to perform its loading upon a file save.

**Comment by webdesign4alll** [comment]
I have changed the workspace to a particular folder yet its not working. Its working well on linux with auto save but not working on windows...

**Comment by im-rahulkgupta** [comment]
> Can anyone help me please


> ## I'm submitting a...
> ```
> [x] Regression (a behavior that used to work and stopped working in a new release)
> [x] Bug report  
> [ ] Feature request
> [x] Documentation issue or request
> 
> ## Current behavior
> 
> live server not reloading automatically
> 
> ## Expected behavior
> 
> it reload without manually reloading the browser
> 
> ## Environment
> 
> ```html
> Browser:
> 
> - [x] Chrome (desktop) version 
> - [x] Firefox version 
> 
> For Tooling issues:
> 
> - Live Server: 5.6.1
> - Platform:  Windows
> - Visual Studio Code: 1.33.1
> ```

What is the solution for this ?

**Comment by Enio11** [comment]
> > Can anyone help me please
> 
> > ## I'm submitting a...
> > ```
> > [x] Regression (a behavior that used to work and stopped working in a new release)
> > [x] Bug report  
> > [ ] Feature request
> > [x] Documentation issue or request
> > 
> > ## Current behavior
> > 
> > live server not reloading automatically
> > 
> > ## Expected behavior
> > 
> > it reload without manually reloading the browser
> > 
> > ## Environment
> > 
> > ```html
> > Browser:
> > 
> > - [x] Chrome (desktop) version 
> > - [x] Firefox version 
> > 
> > For Tooling issues:
> > 
> > - Live Server: 5.6.1
> > - Platform:  Windows
> > - Visual Studio Code: 1.33.1
> > ```
> 
> What is the solution for this ?

I think it was the VSCode Update, cuz until then mine was working normally after updating

**Comment by webdesign4alll** [comment]
I have a solution : 1. create a seperate js file
* 2.  write the js code to reload html page automatically ** onchange or **ontimeinterval 
3. I am using it currently like this
4. Bye bye live server

**Comment by DeeperAuto2335** [comment]
In my case just change de autoSave work.
Just change the configuration for afterDelay
![image](https://user-images.githubusercontent.com/53350158/120082620-4deff480-c09a-11eb-898c-86b93580cd86.png)

**Comment by JeanMesa** [comment]
> Can anybody please answer me???

Hey! I ran into this issue today and here's what solved it for me. I'm assuming you're on OSX? If so, your shell was recently changed from bash to zsh. If this the case for you, you have to manually select a new terminal profile in vscode (zsh if available) or create one if unavailable. Here's how to get to those settings: CMD + Shift + P to open command in vscode (not terminal) then type in "terminal profile" and select "Select Default Profile". This will provide a list of terminal profiles recognized at that point by vscode. If zsh is NOT there, escape and open command palette again, type in terminal profile but this time selecting "create new integrated terminal (with profile)". From here create your zsh profile and set it as your default. Once done, go into your settings.json file and either remove or comment out the line that sets your default profile as this feature is now deprecated in newer versions of vscode and will not work and throw you a "PROBLEM" in vscode, nothing serious only severity 4 but it's really annoying. Anyway, please lmk if this helps! :)

**Comment by Dwarageshg** [comment]
> ## I'm submitting a...
> ```
> [x] Regression (a behavior that used to work and stopped working in a new release)
> [x] Bug report  
> [ ] Feature request
> [x] Documentation issue or request
> 
> ## Current behavior
> 
> live server not reloading automatically
> 
> ## Expected behavior
> 
> it reload without manually reloading the browser
> 
> ## Environment
> 
> ```html
> Browser:
> 
> - [x] Chrome (desktop) version 
> - [x] Firefox version 
> 
> For Tooling issues:
> 
> - Live Server: 5.6.1
> - Platform:  Windows
> - Visual Studio Code: 1.33.1
> ```

i had a same problem and got fixed https://www.youtube.com/watch?v=ivmcRXzGAEk 
see this video for help

**Comment by markymauro13** [comment]
yeah this is not working on my desktop. However, it does seem to be working on my laptop.

**Comment by cmalpicag** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

This solution is worked for me. Thanks.

**Comment by KR-blip** [comment]
enable the autosave
just go to file-> autosave

**Comment by alextechtx** [comment]
This has to be related to updates in VS Code causing random hiccups after being updated. While reading this, the only change I did was re-type my UTF-8, save and relaunch (Go Live) again and it started to work again. It was lowercase so I thought re-typing it as uppercase may help and it started working again. Now, 100% sure the uppercase vs. lowercase has nothing to do with it since I changed back to lower case and it is still working, but retyping, saving and relaunching it did trigger something that made it work.

Note: I did not try to reinstall, reset, nor attempted to create a new file. Would have been good troubleshooting, but fortunately did not have to spend much time on it.

**Comment by kzamojcin** [comment]
So... few things work here like a charm, for which I'm really grateful. Yet I have discovered a new one.

Just to recap:
- check that your code is inside `<html>` structure with `<meta charset="UTF-8">` encoding in the `<head>` element
- check naming of your file, it should not start with special signs like "." "#" etc. 

- **and a new one** which I've just discovered is that my .html file was placed on a shared space inside my company, when I have transfered the file into my local "C:\" drive it started to work.. so basically, **keep your files at your local drive**

Cheers o/

**Comment by ujwal49** [comment]
**SOLUTION**
Follow the below steps (which worked for me):

install live server extension in your browser and add the live server URL It reloaded automatically after I saved the file!!!

**Comment by meowalien** [comment]
Looks like the life server will only auto-refresh the full definition HTML file
```
<html lang="en">
	<head>
		<meta charset="UTF-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<title>Document</title>
	</head>
	<body>
           // Your code
	</body>
</html>
```
I fix this by adding the above code

**Comment by frida96** [comment]
click on auto save from file menu or push ctrl+s

**Comment by michellethompson-github** [comment]
> I had the same issue. Until now, i couldn't figure out what's the problem. My environment:
> Windows 10
> Chrome Version: 76.0.3809.100 (Official Build) (64-bit)
> VSCode 1.36.1
> Live Server 5.6.1
> 
> So far, it's been a random occurrence for me when i encounter this "page doesn't reload automatically". I sometimes have it working after i click save, but only after a couple of minutes until the "auto-reload" doesn't work anymore. When i restarted my computer, it's a 50/50 chance that LiveServer will work correctly. Right now, i dont have a reliable solution as what may causing this problem. I think it could be my Windows 10, but i don't know how to check it.

**Comment by eliteops** [comment]
@ritwickdey-kreeti  live server not refreshing on default firefox page .using alt+l+o nothing updated on page. kindly solve this issue. 
![Screenshot (3)](https://user-images.githubusercontent.com/62295367/129607436-fcb92942-8103-40dc-90ed-07dab10da8df.png)

**Comment by KR-blip** [comment]
Check if you have the option for auto save enabled or not. Enable it

**Comment by cmalpicag** [comment]
> @ritwickdey-kreeti live server not refreshing on default firefox page .using alt+l+o nothing updated on page. kindly solve this issue.
> ![Screenshot (3)](https://user-images.githubusercontent.com/62295367/129607436-fcb92942-8103-40dc-90ed-07dab10da8df.png)

https://github.com/ritwickdey/vscode-live-server/issues/452#issuecomment-862546673

**Comment by cmalpicag** [comment]
https://github.com/ritwickdey/vscode-live-server/issues/452#issuecomment-862546673
> @ritwickdey-kreeti live server not refreshing on default firefox page .using alt+l+o nothing updated on page. kindly solve this issue.
> ![Screenshot (3)](https://user-images.githubusercontent.com/62295367/129607436-fcb92942-8103-40dc-90ed-07dab10da8df.png)

**Comment by eliteops** [comment]
@KR-blip Auto save already enabled and working fine , @cmalpicag  useWebExt already unchecked ,but  problem is, I have to reload the firefox browser page manually to see changes.

**Comment by tahmidmahi04** [comment]
> **SOLVED**
> 
> In my case:
> I had my HTML file inside a folder, which name starts with a period `.`
> 
> Don't do this 👇
> 
> > http://127.0.0.1:5500/.folder/index.html

Life Saver Solution!!

**Comment by kshitij-agarwal** [comment]
Please use the proper HTML syntax which is 👇

```html
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<title>Hello, World</title>
</head>
<body>
	My first Web dev code
</body>
</html>
```

Being a newbie, I was only using the **`head tag`**. The web page was working in the browser but the live server couldn't process it. You can get the above snippet in vscode by typing `html:5` and then hitting tab.

**Comment by philharlow** [comment]
As a hold-over from using serve, I was running `live-server --port=3000 ./` to serve the local directory. Removing the `./` fixed it for me

**Comment by mikearthur7b0** [comment]
For me, the problem was the line
`<script src="src1.js"/>`
Live update worked when I changed it to
`<script src="src1.js"></script>`
Be sure to also follow the comments above to set up your `<html>` and `<head>` correctly:
`<html lang="en">
	<head>
		<meta charset="UTF-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<title>Document</title>
	</head>
    <body>
        <h1>Welcome!</h1>
    </body>
</html>`

**Comment by kokitkarak** [comment]
Just uncheck the option "Use Web Ext". close your visual studio code. Open it again and your live server works fine by auto reloading.

**Comment by MarceloHcb** [comment]
> **RESOLVIDO**
> 
> No meu caso: eu tinha meu arquivo HTML dentro de uma pasta, cujo nome começa com um ponto`.`
> 
> Não faça isso apontar para baixo
> 
> > http://127.0.0.1:5500/.folder/index.html

Thanks

**Comment by juliostricker** [comment]
Eu tinha uma meu arquivo .html dentro de uma pasta cujo o nome começava com o caractere "#" e por conta disso o LiveServer não estava conseguindo atualizar automaticamente.
-------------------------
I had my .html file inside a folder whose name started with the character "#" and because of that the LiveServer wasn't able to update automatically.

**Comment by Khalid-Sama** [comment]
> ## I'm submitting a...
> ```
> [x] Regression (a behavior that used to work and stopped working in a new release)
> [x] Bug report  
> [ ] Feature request
> [x] Documentation issue or request
> 
> ## Current behavior
> 
> live server not reloading automatically
> 
> ## Expected behavior
> 
> it reload without manually reloading the browser
> 
> ## Environment
> 
> ```html
> Browser:
> 
> - [x] Chrome (desktop) version 
> - [x] Firefox version 
> 
> For Tooling issues:
> 
> - Live Server: 5.6.1
> - Platform:  Windows
> - Visual Studio Code: 1.33.1
> ```



> In my case just change de autoSave work. Just change the configuration for afterDelay ![image](https://user-images.githubusercontent.com/53350158/120082620-4deff480-c09a-11eb-898c-86b93580cd86.png)

29/11/2021
This worked perfectly, Thank you!. You all should check this out.

**Comment by julianwachholz** [comment]
HTML changes will not be detected when the folder name is `html`. I had an `index.html` that wouldn't update its contents. It worked once I renamed the folder or moved the file somewhere else.

**Comment by jmsanchezfin** [comment]
> Looks like the life server will only auto-refresh the full definition HTML file
> 
> ```
> <html lang="en">
> 	<head>
> 		<meta charset="UTF-8">
> 		<meta http-equiv="X-UA-Compatible" content="IE=edge">
> 		<meta name="viewport" content="width=device-width, initial-scale=1.0">
> 		<title>Document</title>
> 	</head>
> 	<body>
>            // Your code
> 	</body>
> </html>
> ```
> 
> I fix this by adding the above code

This solved the issue

**Comment by LeoCosta7** [comment]
> For me, the problem was the line `<script src="src1.js"/>` Live update worked when I changed it to `<script src="src1.js"></script>` Be sure to also follow the comments above to set up your `<html>` and `<head>` correctly: [#452 (comment)](https://github.com/ritwickdey/vscode-live-server/issues/452#issuecomment-925845096)

This solved my problem too. Thank's man

**Comment by justaprogrammer96** [comment]
> ## I'm submitting a...
> ```
> [x] Regression (a behavior that used to work and stopped working in a new release)
> [x] Bug report  
> [ ] Feature request
> [x] Documentation issue or request
> 
> ## Current behavior
> 
> live server not reloading automatically
> 
> ## Expected behavior
> 
> it reload without manually reloading the browser
> 
> ## Environment
> 
> ```html
> Browser:
> 
> - [x] Chrome (desktop) version 
> - [x] Firefox version 
> 
> For Tooling issues:
> 
> - Live Server: 5.6.1
> - Platform:  Windows
> - Visual Studio Code: 1.33.1
> ```

Make sure to put: <meta charset="UTF-8" /> in the code!

**Comment by VictorCeSilva** [comment]
> enable the autosave just go to file-> autosave

Thanks. YOU SAVE ME! I LOVE YOU

**Comment by NikitaLys** [comment]
enable the autosave just go to file-> autosave

**Comment by krisrana** [comment]
A simple fix to make sure the live server auto-refreshes is to add HTML boilerplate as such.
`
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    
</body>
</html>`

**Comment by MCGirgin** [comment]
You must press `ctrl + s` to save before closing visual studio.

**Comment by Andresfgomez970** [comment]
In the hope it could help someone I describe how I solve my problem.

In my case I got the warning ["Visual Studio Code is unable to watch for file changes in this large workspace"](https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc). I realized that I had left open a big project in another Visual Studio Code window (window 1), so it was not enough to simply close and open the window (window 2) in which I was editing the html. I close all windows and  reopen, and now it is working perfectly.

**Comment by divrawat** [comment]
When I am saving my code and responsive mode is on live server just get disconnected but it is working properly when my screen is not in responsive mode. any Answers ???

**Comment by priyanshunegi17** [comment]
For me, I didn't closed the script tag properly. now it's working fine.

**Comment by dariormaracaja** [comment]
For me it was enough to click on "Go live" in the footer of the project
![Capturar](https://user-images.githubusercontent.com/78767804/156946793-14e3c2c9-7052-41db-8b5a-1d5438266103.PNG)

**Comment by KMcKraken** [comment]
After trying all suggestions, I finally had to do a complete reinstall of VSCode. This means wiping the software, which alone didn't work, wiping the extensions, which alone didn't work, and finally deleting the entire "%appdata%\code" folder. Only after this last step did the Live Server start working again for me. Seems there may have been come conflicts going on somewhere in there,

Now, I don't know what all is contained in that folder, but I'm personally okay with starting from scratch. Some may not be, so be careful what you delete.

This is just my experience.

**Comment by wildermiranda** [comment]
My folder name was with the character " ' ". After I changed the name of the folder, the live server started working normally again. Hope it helps!

**Comment by nickmura** [comment]
> Was having the same issue. Looks like it was as simple as how I was opening the file. When I go to File > Open Folder, it seems to work every time. When I open the individual files, it doesn't work. This thread was the key, Thank you!

UTF-8 charset was already set, this solved my issue.

**Comment by BenM61** [comment]
might worth to mention- make sure the tags in your file are closed. I had this problem because of a open textarea tag that canceled the rest of the body

**Comment by agabler** [comment]
I went though all above setting options and advices and now it works. Thank you all! :) I spent 3 hours looking for solution!


So you should take all these steps:

-change settings in VS Code. If you go to Settings, search for these options and set them as below:

Live Server Settings: 
Custom Browser: chrome (I use chrome, you can choose yours of course)
Full Reload:  yes, you should mark it
Port: 0
Use Web Ext: unmark it
Wait: 3

Settings:
autosave: afterDelay

-check if your HTML file name is correct (without special signs etc) and if there is utf-8 in <head></head>
-I also installed extension in my Chrome browser ("LiveReload")- you can download it in GoogleStore.

Save all changes, reopen chrome and VsCode and check if it works.

**Comment by JanAntuan99** [comment]
in case someone still looking for answer:
we dont need any sxtra setting or app. Just live server and VS,
in VS go to setting and serach for "Auto Save" and change it to "afterDelay" than search for Auto save Delay and typ there "0" or "1".
now its Working! hoho

p.s: Json file in case you are familiar with:

    "files.autoSave": "afterDelay",
    "files.autoSaveDelay": 1,

**Comment by Kannu-Mandora** [comment]
Bro, I just fixed this issue. Go to vs-code settings and search autosave option then, select **_"AfterDelay"_** then your file automatic save and changes occur in live server shown. 
**Thanks me later.** 😊😊

**Comment by Jasper-Rupia** [comment]
index.html works fine. but how do i reload any file other than index.html, 
let say myfile.html doesint work on me.

**Comment by SPARTAKRAGA** [comment]
in my case the problem was Live Server > Settings : Full reload (checked)

**Comment by phamthanggg** [comment]
.folder-name won't working. remove . before folder-name. it's success for me

**Comment by msunsmiley** [comment]
None of what people are suggesting is working for me. I have to manually refresh my page

**Comment by preciousBlessed** [comment]
Thanks @tahmidmahi04 ... you're a genius. I was in a folder like named like this: .vscode  ...I guess the dot mean something. Once I created a new folder without a dot starting it, code started updating on its own ... life    ...smooth... no stress... life Continues... Bless you son!

**Comment by darrienjohnson** [comment]
> Bro, I just fixed this issue. Go to vs-code settings and search autosave option then, select **_"AfterDelay"_** then your file automatic save and changes occur in live server shown. **Thanks me later.** 😊😊

Thank you so much

**Comment by IrumiDesu7** [comment]
> For me, the problem was the line `<script src="src1.js"/>` Live update worked when I changed it to `<script src="src1.js"></script>` Be sure to also follow the comments above to set up your `<html>` and `<head>` correctly: [#452 (comment)](https://github.com/ritwickdey/vscode-live-server/issues/452#issuecomment-925845096)

This one solved it for me. I literally just change the script to be a self closed element and was so confused why the auto reload isn't working anymore. Turns out the script need a closing tag for it to work

**Comment by franklinejoe** [comment]
Here's the simple solution that worked for me
Go to settings 
Go to search settings box
Search for autosave, turn it on and set it to 'after delay'
Search for autosave delay and set it to a very small value( mine was '1' and it's very fast in reloading and updating my work automatically on the browser )
That's all it took me and mine is working perfectly

**Comment by pacoduabe** [comment]
For me it worked just by clicking on File>>Auto Save

**Comment by Sarkoz1** [comment]
i switched to Live Server (Five Server) extension, it works perfectly without any tweaking

**Comment by mariusz1802** [comment]
I tried every solution from this topic and another websites and nothing works. How Sarkoz1 said.  Best solution can be switch to five server and it's works for me too. Make sure that your script tag has </script> addiotion tag. With single <script />  five server doesn't see js file.

**Comment by deoxify** [comment]
Tried everything here, completely removing and reinstalling code, no luck.
Then I noticed on the status bar that my html file was somehow encoded in UTF-16, tried saving it with UTF-8 and it started working. Good luck with yours :P

**Comment by hboppe** [comment]
I found this solution as a review of Live server web extension on google chrome:  

Well, here is something that may help some people who are struggling setting up this extension. 

1. Install live server extension on VS Code
2. Install browser extension
3. Install PHP server if needed and make sure it's running before going to next steps
4. Open VS Code and then click on Go Live at the bottom of VS Code
5. Once you clicked on go live, it will open a new tab on your browser. Now copy the address showing in your browser address bar. It will be something like this (http://127.0.0.1:3000/), this address may vary, depend on your settings. Okay, now click on the browser extension and add this whole address (including HTTP) under the Liver Server Address on browser extension. Also, make sure you have the correct folder path to your project under the Actual Server Address on browser extension, before saving the settings in browser extension. 
6. Now restart your browser, VS Code, open the file in VS Code to edit before hitting the Go Live button, and then click on Go Live at the bottom of VS Code
7. The page it will open, it will not be the correct page so, now type in your local server address (path to the project folder such as... http://localhost/myproject/) manually and type the file name at the end such as http://localhost/myproject/index.php or which ever file you want to edit. Then hit enter, and it will open the file and now when you update your file in VS Code, it will update live in your browser too. 
I know, we should not have to enter the file manually and would be nice VS Code should open it automatically but, it's still better than refreshing the page all the time. By the way if someone else has any better info then please share with all of us and make the life easier. Also, thanks to this extension developer who has come up with this great idea!

**Comment by hboppe** [comment]
> 

Try the solution I just added above

**Comment by The-Deepest-Net** [comment]
@hillarionCodes Are you opening vscode into a remote window like linux server or WSL?

**Comment by Henroast** [comment]
apparently what it was for me is the page need <p></p> on it for a paragraph even if you have nothing in there yet and it solved my problem, just do the ! start and add a paragraph that way u are also in UTF-8.

**Comment by Jacobkyalo** [comment]
i tried five server extension and it worked thanks @Sarkoz1

**Comment by deakisti** [comment]
> > It only refreshes after I save the doc for me.
> 
> Had the same issue. Was solved by changing autosave option in setting to afterDelay.

thx, this worked for me :)

**Comment by Lucas-dev1** [comment]
> **SOLVED**
> 
> In my case: I had my HTML file inside a folder, which name starts with a period `.`
> 
> Don't do this point_down
> 
> > http://127.0.0.1:5500/.folder/index.html

same here, name folder initiate with "#".

**Comment by pulkit-sharma-22** [comment]
> I had the same issue and my problem was that my html file wasn't encoded in utf8

THANKS! That was my issue as well. Now it's working. THANKS!

**Comment by walkter** [comment]
> i switched to Live Server (Five Server) extension, it works perfectly without any tweaking

@Sarkoz1 Thank you! Thank you! Thank you!
It worked! Simple and fast.

**Comment by Patrick-xtronics** [comment]
> > Ele só atualiza depois que eu salvo o documento para mim.
> 
> Teve o mesmo problema. Foi resolvido alterando a opção de salvamento automático na configuração para afterDelay.

Funcionou, Valeu!!!

**Comment by alanbernali** [comment]
If live server still doesn't working for you, you can install **Live Preview extension** by microsoft. That extension led you preview the change in your vscode and browser. It work well for me.

**Comment by BrianHHough** [comment]
In case this helps anyone, this is what fixed it for me:
- Press [CMD] + [SHIFT] + [P]  to get the settings dropdown
- Start typing "Live Server" and then in the drop down items, select: `Live Server: Change Live Server Workspace` (should be the first one)
- This should provide a drop down of your project's repo/folder name, select this
- Continue writing to your project
- Repeat this process if the save / reload stops working on your Live Server

There must be something about refreshing the Live Server for an individual project that perhaps VS Code's recent update gets confused with?

Hopefully this helps anyone who's stuck 🙏

![Screen Shot 2023-03-14 at 6 10 02 PM](https://user-images.githubusercontent.com/56947075/225153635-8319cf19-7bb5-41ae-aabe-a6f8d1b6efd5.png)

**Comment by pratiksharan** [comment]
> I just found this, and then randomly at the same time I noticed a notification in vscode telling me that live reload isn't possible without `head` and `body` (I was editing a file that's just an html fragment)

OMFG THANK U SO MUCH! I LUV U!

**Comment by slashedzer0** [comment]
> Make sure meta charset = "UTF-8" is in the head.

this one worked for me

**Comment by josh-ulrich** [comment]
just did this an works like a charm.....

• Open settings.
• Search for 'autosave'.
• Set it to 'afterDelay'.
• Just below that, set the AutoSaveDelay to anything very less like 1 upto 5
• Now search for 'Wait' in settings.
• In LiveServer > Settings:Wait, set the value low to like 3
This should help

**Comment by b4z1z** [comment]
> **SOLVED**
> 
> In my case: I had my HTML file inside a folder, which name starts with a period `.`
> 
> Don't do this 👇
> 
> > http://127.0.0.1:5500/.folder/index.html

thx men u just save me ur the best don t forget disable the point and the SPACE in the folder

**Comment by minashaker77** [comment]
> For me it worked just by clicking on File>>Auto Save

finally it works for me, thank you a lot!

**Comment by madbag** [comment]
> Here's the simple solution that worked for me Go to settings Go to search settings box Search for autosave, turn it on and set it to 'after delay' Search for autosave delay and set it to a very small value( mine was '1' and it's very fast in reloading and updating my work automatically on the browser ) That's all it took me and mine is working perfectly

Thanks! This worked for me.

**Comment by Shan3774** [comment]
> I had the same issue, thank you!

**Comment by Shan3774** [comment]
I had the same issue as  @b4z1z , html file was inside a folder that started with a period, renaming the folder fixed it

**Comment by Shan3774** [comment]
> > **SOLVED**
> > In my case: I had my HTML file inside a folder, which name starts with a period `.`
> > Don't do this point_down
> > > http://127.0.0.1:5500/.folder/index.html
> 
> thx men u just save me ur the best don t forget disable the point and the SPACE in the folder

I had the same issue, thank you !

**Comment by skyisthelimpet** [comment]
> I was having the same issue.
> 
> It seems to work fine if I open _a specific file_ with Live Server by either the `Ctrl-Shift-P` Command Pallet, Right-click menu, or `Alt-L Alt-O` key command. It doesn't work properly if I start Live Server by clicking "Go Live" in the status bar at the bottom.
> 
> Hope this helps someone!

Thank you. Actually solved my problem.

**Comment by MraidiAchref** [comment]
Just try to hit the auto save button , so that vscode save every change automatically and feed back the live server to refresh himself
![Capture d'écran 2023-08-12 123258](https://github.com/ritwickdey/vscode-live-server/assets/140330570/1f27237e-2f7b-4880-bbc2-4fa2dafe57a7)

**Comment by Abdelmalek-Aouissa** [comment]
Not sure is this helps, but people tend to over look the radio icon at the bottom right of VS Code, called "Go Live", after installing/reinstalling Live Server. Have you tried that?

**Comment by magvtv** [comment]
I realized one has to go to file then press 'auto save' then when you code, switching to the browser you will find that it automoatically does the rerendering

**Comment by jbdr1403** [comment]
> Can anybody please answer me???

Always make sure your HTML document have a boiler plater 

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body></body>
</html>

**Comment by jbdr1403** [comment]
> Can anybody please answer me???

Always make sure your HTML document have a boiler plater 

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body></body>
</html>

**Comment by Abdallahossama** [comment]
guys I tried everything but nothing worked with me any help ?

**Comment by Drup1k** [comment]
It also stopped working for me. 
Tryed all off the above...

**Comment by Abdallahossama** [comment]
I fix it .... it was problem of slow hard disk just move the project files for faster hard disk it woks

**Comment by MarkTRoss** [comment]
-Abdallahossama   Kinda sounds not right but I won't completely discount it.
As a note, everyone should be using SSD these days, but you might meant that.

**Comment by pettinelli1202** [comment]
If the problem in Live Server not reloading automatically, my config is ubuntu 22 and brave browser...

The live server in ritwickdey VS Code Live Server is not reloading automatically, requiring manual intervention to reload the browser.

My solution:
To resolve this issue, you can try adjusting the autosave and wait settings in ritwickdey VS Code Live Server. 

1 -> Open the settings, search for ‘autosave’ and set it to ‘afterDelay’, set the AutoSaveDelay to a low value such as 5;
2 -> Search for ‘Wait’ in the settings, and set the value of LiveServer > Settings:Wait to a low value such as 3. These steps should help resolve the issue of the live server not reloading automatically;

**Comment by mhvelplund** [comment]
> Works on my work computer, not my personal computer. , refreshed, reloaded, uninstalled & re-installed several different ways. Still not auto-refreshing. Console says "live reload enabled"... I've narrowed my open folders to just one,...

This worked for me too. Weird. Thanks for the top :)

**Comment by osldgoth** [comment]
> > **SOLVED**
> > In my case: I had my HTML file inside a folder, which name starts with a period `.`
> > Don't do this point_down
> > > http://127.0.0.1:5500/.folder/index.html
> 
> same here, name folder initiate with "#".
Windows 10
Firefox: 122.0
vs code: 1.86.1

I too named my folder beginning with a "#" (ex: #foobar) the browser refresh did not work.
after renaming the folder (ex: foobar) the browser refresh began to work again.

I tested other characters "~ ` ! @ $ % ^ & ) ( , _ - + = ;" except for square and curly brackets and single quote.
 "." and "#" are the only ones where the browser refresh fails to work as far as I know

**Comment by Santzu-27** [comment]
###  **SOLVED MY PROBLEM** - It was within HTML **CODE** ""error"" 

I've tried all of those solutions above and none worked, but, there is this single comment that speaks about how that for him, the problem was that his Live Server didn't update when the html had code errors, however, my html didn't showed up errors, 
**_BUT_**, 
in my content, i had a paragraph with a code fragment, which put it in `<xmp>` tag, AND THEN, inside of **THIS** code (which wasn't actually part of my code), my vscode was signaling error. I commented it this part and my Live Server started to update.

**Comment by nevergu69** [comment]
> There's one thing that resolves the issue. • Open settings. • Search for 'autosave'. • Set it to 'afterDelay'. • Just below that, set the AutoSaveDelay to anything very less like 5 (I have set it to 1) • Now search for 'Wait' in settings. • In LiveServer > Settings:Wait, set the value very low (e.g. 3). This should help regardless of platforms :)

awesome , thankyou

**Comment by NearlyNormal** [comment]
None of this works. 

Mac, latest OSX, latest VSCode, latest Live Server. 

- HTML has meta utf8 and head and body. 
- All the live server settings are fine. 
- Default browser is Brave (chrome under the hood). 
- Doing Command L Command O opens the browser just fine. But no live reload. 

What else? Long thread and no solution.

**Comment by Aditya234892** [comment]
Same thing was happening to me, there was some problem in settings.json file of live server I replaced all of the content of my live server's json file with this:-

{
    "editor.mouseWheelZoom": true,
    "editor.multiCursorModifier": "ctrlCmd",
    "editor.fontSize": 20,
    "files.autoSave": "afterDelay",
    "files.autoSaveDelay": 500,
    "liveServer.settings.donotShowInfoMsg": true,
    "liveServer.settings.useWebExt": false,
    "liveServer.settings.ChromeDebuggingAttachment": false,
    "liveServer.settings.port": 5500,
    "liveServer.settings.donotVerifyTags": true,
    "liveServer.settings.fullReload": true,
    "liveServer.settings.wait": 100,
    "liveServer.settings.AdvanceCustomBrowserCmdLine": "",
    "liveServer.settings.ignoreFiles": [
        ".vscode/**",
        "**/*.scss",
        "**/*.sass",
        "**/*.ts"
    ],
    "liveServer.settings.root": "/"
}

There is not much change in json file, just some orders are changed and some lines are added.

It started working for me.

Hope it helps. ✨

**Comment by briankro951** [comment]
> "liveServer.settings.useWebExt": false,

I had this setting set to true (Both Chrome and FireFox weren't working). After I set this to false, and it started working again. Even works on safari now!

**Comment by kbcode85** [comment]
i think is the windows issue, identical workspace same settings vs code synced, on macbook it works for me and on windows 11 not anymore

**Comment by alsongard** [comment]
Same issue I'm running on kali. I did the same thing as @briankro951 and it worked. Thanks @briankro951.

**Comment by Rishabhgupta95** [comment]
> Same thing was happening to me, there was some problem in settings.json file of live server, I replaced all of the content of my live server's JSON file with this:-
> 
> { "editor.mouseWheelZoom": true, "editor.multiCursorModifier": "ctrlCmd", "editor.fontSize": 20, "files.autoSave": "afterDelay", "files.autoSaveDelay": 500, "liveServer.settings.donotShowInfoMsg": true, "liveServer.settings.useWebExt": false, "liveServer.settings.ChromeDebuggingAttachment": false, "liveServer.settings.port": 5500, "liveServer.settings.donotVerifyTags": true, "liveServer.settings.fullReload": true, "liveServer.settings.wait": 100, "liveServer.settings.AdvanceCustomBrowserCmdLine": "", "liveServer.settings.ignoreFiles": [ ".vscode/**", "**/_.scss", "**/_.sass", "**/*.ts" ], "liveServer.settings.root": "/" }
> 
> The changes to the JSON file includes minor modifications like: 1.Reordering of the code elements. 2.Alterations to some values. 3.Addition of a few extra lines.
> 
> It started working for me.
> 
> Hope it helps. ✨

Thankyou so much!! This also worked for me.

**Comment by edooardo** [comment]
My browser was updating but not fully… 
So, checking "Full reload" under Live Server Config worked for me. 
Or also, in json file: "liveServer.settings.fullReload": true

Hope it also helps.

**Comment by murs313** [comment]
In my case, only the HTML auto-reload was not working.
This problem was resolved by checking the `Live Server > Settings: Donot Verify Tags` option.
Or, you can add this line to your settings file:
```
"liveServer.settings.donotVerifyTags": true
```

**Comment by pulkit-sharma-22** [comment]
> **SOLVED**
> 
> In my case: I had my HTML file inside a folder, which name starts with a period `.`
> 
> Don't do this 👇
> 
> > http://127.0.0.1:5500/.folder/index.html

THANK YOU!! This worked. I named my root folder ../#001/index.html

removed the # and the auto refresh on ctl+S started working like usual. Thank you so much for sharing this!
